### PR TITLE
Prevent missing locale from crashing snippet import

### PR DESCRIPTION
**Pull request by hanneswernery**
### 1. Why is this change necessary?

Right now if you import snippets to the db while installing a plugin and if a specific entry in `s_core_locales` is missing, the whole import will crash with the following exception `Call to a member function getId() on null` in [this line](https://github.com/shopware/shopware/blob/d4c4f2e731d94c9a385791fc5898e9c3c4a18479/engine/Shopware/Components/Snippet/DatabaseHandler.php#L138) and the plugin installation will fail.
We encountered multiple customers who _cleaned up their database_ by removing some locales. Which is bad practise by itself, but it also causes all of our plugin installations to fail.

### 2. What does this change do, exactly?

This PR adds a safeguard that checks if the requested locale does in fact exist before importing a snippet for it. If no locale was found the specific snippets will be skipped. This way all remaining (and working) snippets can still be imported normally, which is better than throwing an exception (which also prevents the plugin from being installed) and probably what mentioned customers want to happen.

### 3. Describe each step to reproduce the issue or behaviour.

If you delete a locale that is needed in the snippets of a plugin from the db and try to import snippets of said plugin, it will fail for all remaining snippets of this plugin.

### 6. Checklist

- [X] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Components/Snippet/DatabaseHandler.php [hunk 1]
```diff
@@ -135,9 +135,12 @@ public function loadToDatabase($snippetsDir = null, $force = false, $namespacePr
                     $locale = $localeRepository->findOneBy(['locale' => $index]);
                 }
 
-                $databaseWriter->write($values, $namespacePrefix . $namespace, $locale->getId(), 1);
+                // Only write entry if locale was found
+                if (!$locale) {
```
Comment (mitelg) [inline engine/Shopware/Components/Snippet/DatabaseHandler.php]: shouldn't this be `if ($locale) {` ?

**Review by mitelg — CHANGES_REQUESTED** [review]

File: engine/Shopware/Components/Snippet/DatabaseHandler.php (hunk 1 above)
Comment (hanneswernery) [reply engine/Shopware/Components/Snippet/DatabaseHandler.php]: Of course it should. Thanks!

**Review by mitelg — APPROVED** [review]

**Comment by hanneswernery** [comment]
I've added a test that fails without the my changes in the `DatabaseHandler.php`.

**Comment by cjuner** [comment]
@hanneswernery, please rebase for Shopware 5.5, or delegate it appropriately.

**Comment by soebbing** [comment]
Hi @hanneswernery, thank you very much for your contribution! 🎉

**Pull request closed without merging**